File: java/L1100/LeetCode1128.java
package L1100;

import java.util.HashMap;
import java.util.Map;

public class LeetCode1128 {
    public int numEquivDominoPairs(int[][] dominoes) {
        HashMap<String,Integer> map = new HashMap<>();
        for (int[] dominoe : dominoes) {
            String key = (dominoe[0] <= dominoe[1]) ? dominoe[0] + ":" + dominoe[1] : dominoe[1] + ":" + dominoe[0];
            map.put(key, map.getOrDefault(key, 0) + 1);
        }
        int ans = 0;
        for (Map.Entry<String,Integer> entry : map.entrySet()) {
            int v = entry.getValue();
            ans += v * (v - 1) / 2;
        }
        return ans;
    }
}

File: cpp/L1100/LeetCode1128.cc
#include <unordered_map>
#include <utility>
#include <vector>
using std::vector;

class Solution {
public:
    int numEquivDominoPairs(vector<vector<int>>& dominoes) {
        std::unordered_map<int, int> mp;
        int ans{0};
        for (vector<int> const &d : dominoes) {
            int x = d[0];
            int y = d[1];
            if (x > y) {
                std::swap(x, y);
            }
            int xy = x * 10 + y;
            ans += mp[xy];
            mp[xy] += 1;
        }
        return ans;
    }
};